src/main/java/com/google/sps/utils/GameUtils.java

```diff
@@ -42,32 +43,32 @@ public static Entity createGameEntity(String gameName, DatastoreService datastor
     
     // intializing game property values
     Entity gameEntity = new Entity("Game");
+    Game game = new Game(gameEntity);
     ArrayList<String> userIds = new ArrayList<String>();
     String quizQuestion = "";
     long quiz_timestamp = 0;
- 
-    gameEntity.setProperty("gameName", gameName);
-    gameEntity.setProperty("userIds", userIds);
-    gameEntity.setProperty("quizQuestion", quizQuestion);
-    gameEntity.setProperty("quiz_timestamp", quiz_timestamp);
     Key gameKey = gameEntity.getKey();
- 
+
+    game.setGameName(gameName);
+    game.setUserIds(userIds);
+    game.setQuizQuestion(quizQuestion);
+    game.setQuizTimestamp(quiz_timestamp);
+
     datastore.put(gameEntity);
 
     // getting the game id for the game id property
-    Entity entity = null;
     try{
-      entity = datastore.get(gameKey);
+      gameEntity = datastore.get(gameKey);
 
       String key = KeyFactory.keyToString(gameKey);
-      entity.setProperty("gameId", key);
-      datastore.put(entity);
+      game.setGameId(key);
+      datastore.put(gameEntity);
```

Comment:
You should try to have online one, max two, lines of code in a try block

Reply:
Also aren't we putting try catch blocks that test for null in the test classes?

Reply:
I have this one here because this is where the potential null would be created (as opposed to be created elsewhere and passed in), so I'm checking for it here

Reply:
Okay but can we still reduce it to be only one-two lines in the try block?

Reply:
Done, sorry!